### PR TITLE
Fix late static binding calls for first class callable

### PR DESCRIPTION
**Pull request by mvorisek**
followup to #3361

dedup + fix first class callable

### CODE REVIEW

File: tests/PHPStan/Analyser/nsrt/static-late-binding.php [hunk 1]
```diff
@@ -112,6 +112,15 @@ public function foo(): void
         assertType('bool', $v);
         $clUnioned->outStaticConst($v);
         assertType('bool', $v); // should be bool|int
+
+        assertType('StaticLateBinding\A', A::retStatic(...)());
+        assertType('StaticLateBinding\B', B::retStatic(...)());
+        assertType('Closure', self::retStatic()(...)); // should be static(StaticLateBinding\B)
```
Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/nsrt/static-late-binding.php]: This shows this doesn't really work well.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Analyser/nsrt/static-late-binding.php (hunk 1 above)
Comment (mvorisek) [reply tests/PHPStan/Analyser/nsrt/static-late-binding.php]: fixed, typo on my side in the test

Comment (mvorisek) [reply tests/PHPStan/Analyser/nsrt/static-late-binding.php]: @ondrejmirtes can this PR be merged?

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**